lnwallet/channel.go

```diff
@@ -3092,6 +3283,89 @@ func (lc *LightningChannel) createCommitDiff(
 	}, nil
 }
 
+// getDanglingRemoteUpdates returns all remote log updates that we haven't
+// signed for yet ourselves.
+func (lc *LightningChannel) getDanglingRemoteUpdates() []channeldb.LogUpdate {
+	// First, we need to convert the funding outpoint into the ID that's
+	// used on the wire to identify this channel.
+	chanID := lnwire.NewChanIDFromOutPoint(&lc.channelState.FundingOutpoint)
+
+	// Fetch the last remote update that we have added to our remote
+	// commitment chain.
+	chainTail := lc.remoteCommitChain.tip()
+	lastCommited := chainTail.theirMessageIndex
+
+	// We'll now run through the remote update log to locate the items that
```

Comment:
This is already ensured, because this function is only called when we've signed for all pending updates. I mentioned this in the comment below.

But to prevent any mistakes when using this function in the future, I've added the check. At the moment, it is a no-op though.

Reply:
I am wondering, suppose we would have used two separate db transactions for sign and ack. I think we would then store all updates that they signed regardless of what we acked already? Then also the name `unsignedAckedUpdates` wouldn't be appropriate. It doesn't matter with the current implementation, but if we want to make things 'more correct' maybe we should?

Reply:
If they were separate db transactions, only what is acked matters. We could store their signature+updates, restart and just delete it since we haven't acked yet. They should then resend them.

But yeah, shouldnt matter here, but I think `unsignedAckedUpdates` makes sense since it refers to updates _we_ acked, but _we_ didn't sign.

Reply:
Ok. The spec says this too indeed
> if the commit is not acknowledged (or wasn't sent) the updates are re-sent.

Reply:
So the name is good then.